corehq/apps/reports/standard/users/reports.py

```diff
@@ -122,7 +123,7 @@ def _get_users_es_query(self, slugs):
         )
 
     def _build_query(self, user_ids, changed_by_user_ids, user_property, actions, user_upload_record_id):
-        filters = Q(by_domain=self.domain)
+        filters = Q(for_domain__in=self._for_domains())
```

Comment:
@mkangia this will always filter on all domains associated with the account which means that the only way this is restricted to a single domain is by the user filter. I'm not sure if this is possible but user IDs are provided via URL query params which makes them vulnerable to tampering. 

I think it would make sense to restrict the domain filter to only the current domain except in specific scenarios where we need to expand it (I'm not clear on exactly what the criteria are).

Reply:
@mkangia any thoughts on this?

Reply:
Hey @snopoke 

Sorry for the late response here. I am not up to speed with further progress on this report in USH, so will try to respond from what I remember when I made this change.

> that the only way this is restricted to a single domain is by the user filter

Not sure what you mean here Simon. This is going to filter for changes for all domains in the account. And then we filter further for accessible user_ids. 
There was some follow up being done to update user filters to work for enterprise, which should complete the full set of changes.
Can you share when we would need to restrict to a single domain? 

> user IDs are provided via URL query params which makes them vulnerable to tampering

Won't `EMWF.user_es_query` handle that?

> I think it would make sense to restrict the domain filter to only the current domain except in specific scenarios where we need to expand it (I'm not clear on exactly what the criteria are).

If there are multiple domains, from user experience, we would need to show changes for all domains by default. I don't know if there is a way to provide that differently and I assumed that the account itself would only have a single domain for non-enterprise.